cub/cub/thread/thread_reduce.cuh

```diff
@@ -102,23 +164,50 @@ ThreadReduce(T* input, ReductionOp reduction_op, PrefixT prefix, Int2Type<LENGTH
  *
  * @param[in] reduction_op
  *   Binary reduction operator
- *
- * @param[in] prefix
- *   Prefix to seed reduction with
  */
-template <int LENGTH,
+template <int LENGTH, typename T, typename ReductionOp>
+_CCCL_DEVICE _CCCL_FORCEINLINE // clang-format off
+::cuda::std::enable_if_t<!enable_dpx_reduction<LENGTH, T, ReductionOp>(), T>
+ThreadReduce(T* input, ReductionOp reduction_op) // clang-format on
+{
+  T prefix = input[0];
+  return ThreadReduce(input + 1, reduction_op, prefix, Int2Type<LENGTH - 1>{});
+}
+
+/// Specialization for DPX reduction
+template <int LENGTH, // clang-format off
+          typename T,
+          typename ReductionOp>
+_CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE
+::cuda::std::enable_if_t<enable_dpx_reduction<LENGTH, T, ReductionOp>(), T> // clang-format on
+ThreadReduce(T* input, ReductionOp reduction_op)
+{
+  constexpr auto IS_MIN = ::cuda::std::is_same_v<ReductionOp, cub::Min>;
+  using DpxReduceOp     = ::cuda::std::conditional_t<IS_MIN, DpxMin<T>, DpxMax<T>>;
```

Comment:
Important: C++11 and libcu++ has an internal faster `conditional_t` called `_If`.
```suggestion
  using DpxReduceOp     = ::cuda::std::_If<IS_MIN, DpxMin<T>, DpxMax<T>>;
```